File: week10/week10-6.py
class Solution:
    def largestPerimeter(self, a: List[int]) -> int:
        a.sort(reverse=1)
        n=len(a)
        for i in range(n-2):
            if a[i]<a[i+1]+a[i+2]:return a[i]+a[i+1]+a[i+2]
        return 0

File: week10/week10-6.cpp
class Solution {
public:
    int largestPerimeter(vector<int>& nums) {
        int n=nums.size();
        sort(nums.begin(),nums.end());
        reverse(nums.begin(),nums.end());
        for(int i=0;i<n-2;i++){
            if(nums[i] < nums[i+1] + nums[i+2]) return nums[i] + nums[i+1] + nums[i+2];
        }
        return 0;
    }
};